### PR TITLE
Refactor Redundant Code in ASE/LSE

### LINKED ISSUE #485: Refactor LCC check and n_features_ attribute setting to BaseEmbed
ASE/LSE fit have some redundant code that could be in base

see #462

### PR DESCRIPTION
**Pull request by kareef928**
#### Reference Issues/PRs
Fixes #485 

#### What does this implement/fix? Briefly explain your changes.
Refactors redundant code in ASE/LSE to Base
#### Any other comments?
Simple notebook showing functionality: [Notebook](https://nbviewer.jupyter.org/github/kareef928/graspologic-personal/blob/main/notebooks/embed-refactor.ipynb)

### CODE REVIEW

**Comment by PerifanosPrometheus** [comment]
@kareef928 Kareef, I could peer review this PR once the test errors are fixed.

File: graspologic/embed/base.py [hunk 1]
```diff
@@ -108,7 +113,6 @@ def _pairwise(self):
         """This is for sklearn compliance."""
         return True
 
-    @abstractmethod
     def fit(self, graph, y=None):
```
Comment (loftusa) [inline graspologic/embed/base.py:116]: shouldn't this still be an abstract method?

**Comment by kareef928** [comment]
> @kareef928 Kareef, I could peer review this PR once the test errors are fixed.

Hey @PerifanosPrometheus, sorry for the delay, but the test errors are fixed so you can peer review this PR now

**Comment by PerifanosPrometheus** [comment]
**PR itself (on GitHub)**
- [x] PR has a descriptive and succinct title.
- [x] PR has a brief description of what was changed.
- [x] PR is addressing a change that has already been brought up in an issue. This will ensure that the change you propose is desired by the maintainers, and may provide a chance to discuss implementation details prior to PRing which will save everyone time.
- [x] PR cites any related issues in the PR description and uses [closing keywords](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for any issues that should be closed as a result of the PR.
- [x] PR does not have any extraneous file changes associated with it. One common example of an extraneous change is one that was already made in the target branch, but is showing up as part of this PR under Files changed. (**changes made to rdpg_corr due to bug in testing**)
 - [x] PR does not have any unresolved merge conflicts.
 - [x] If implementing a major new feature or algorithm, a notebook demonstrating the proof of effectiveness is linked in the PR description. Depending on the feature, this kind of validation may take too long to run as a test or in a tutorial notebook.

**Style**

  - [x] Code follows Python variable naming conventions, one description is [here](https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html). Most variables should be snake_case and classes shoud be defined as UpperCamelCase, for example.
  - [x] Code uses descriptive variable names throughout.
  - [x] Code has no commented out "junk code."
  - [x] Line lengths are short (recommended <88 chars). This includes docstrings. Note: [black](https://black.readthedocs.io/en/stable/) tries to shorten line lengths but sometimes it is unable to do so automatically, especially for docstrings.

**Documentation**

 - [x] Any new public classes/functions have docstrings.
 - [x] Any new public classes/functions have been added to the appropriate .rst file [here](https://github.com/microsoft/graspologic/tree/dev/docs/reference) to be rendered by Sphinx.
 - [x] Any major new features (like a new algorithm) are accompanied by a succinct tutorial notebook. Or, if it is agreed upon with the maintainers that this is out of scope for the current PR, a new issue has been created specifying that we need a tutorial notebook for this functionality.
 - [x] Any new tutorial notebooks have been added to the appropriate folder [here](https://github.com/microsoft/graspologic/tree/dev/docs/tutorials) and included in the .rst file [here](https://github.com/microsoft/graspologic/blob/dev/docs/tutorial.rst) to be rendered by Sphinx.
 - [x] Modifications to the documentation render appropriately in the Netlify build.
 - [x] References to parameters in the documentation are wrapped in double backticks (`` on either side of the parameter).
 - [x] References to classes/function/documentation outside of this package use intersphinx to automatically link to the relevant documentation pages.

**Testing**

 - [x] New public classes/functions are tested to ensure they achieve the desired output.
 - [x] New public classes/functions are tested to ensure proper errors are thrown when invalid inputs are passed.

**Comment by PerifanosPrometheus** [comment]
@kareef928 I am not sure whether this PR requires some type of Notebook demonstrating that the changes made did not affect  the implementation results. Looking at the code, it looks fine so if you have already proven that is correctly functioning to Pedigo, I think this PR is ready for final review and eventually merging.

**Review by bdpedigo — APPROVED** [review]
looks good to me. 

@j1c any thoughts? Only hesitation i had in terms of readability was whether its weird for super().fit() to return a graph instead of self which is not sklearny but it also seems pretty clear/useful here

**Comment by j1c** [comment]
> looks good to me.
> 
> @j1c any thoughts? Only hesitation i had in terms of readability was whether its weird for super().fit() to return a graph instead of self which is not sklearny but it also seems pretty clear/useful here

Lets call it `_fit()` instead of `fit()` in the base class, and leave the `fit()` as an abstract method.

**Comment by asaadeldin11** [comment]
@kareef928 see above comment

**Comment by kareef928** [comment]
@bdpedigo @asaadeldin11 can you confirm that the commit I made is what we want?

**Review by asaadeldin11 — APPROVED** [review]
This looks good to me

**Pull request merged**